DEVELOPER_GUIDE.md

```diff
@@ -146,6 +146,7 @@ This repository includes several OpenAPI Specification Extensions to fill in any
 - `x-ignorable`: Denotes that the operation should be ignored by the client generator. This is used in operation groups where some operations have been replaced by newer ones, but we still keep them in the specs because the server still supports them.
 - `x-global`: Denotes that the parameter is a global parameter that is included in every operation. These parameters are listed in the [spec/_global_parameters.yaml](spec/_global_parameters.yaml).
 - `x-default`: Contains the default value of a parameter. This is often used to override the default value specified in the schema, or to avoid accidentally changing the default value when updating a shared schema.
+- `x-distributions`: Contains a list of distributions known to include the API. Use `opensearch.org` for the official distribution, `aos` for Amazon Managed OpenSearch, and `aoss` for Amazon OpenSearch Serverless. 
```

Comment:
I think having `aos` and `aoss` as shortened versions is not immediately grokkable to someone without some AWS specific context. Even as someone with that context having one be a prefix of the other immediately hurts readability. Could we expand these to `amazon-managed` & `amazon-serverless` given we have a full form `opensearch.org`?

We should think ahead for what precedent we set if in future we end up with contributions related to things like [Oracle Cloud Infrastructure Search with OpenSearch](https://www.oracle.com/cloud/search/). (`oci`, `ocisos`, `oracle`, `oracle-managed`, `oci-managed`, etc)

Reply:
Good point.  One thought I had is to use URIs, e.g. `aws.amazon.com/opensearch-service`? WDYT?

Reply:
Could be a bit verbose, AOSS would be `aws.amazon.com/opensearch-service/features/serverless` presumably.

Either way we'd likely need to define an enum somewhere and validate correctness, as spelling mistakes will be a PITA

Reply:
I tried to use `ajv.addKeyword` to define a custom type for the new `x-distribution` options, but couldn't make it work. Maybe @nhtruong can help me make these values check against an enum on top of this PR.

Reply:
We're only using AJV to validate JSON Schemas in the spec. `x-distribution` is not used in a schema, but rather an operation. So it should be validated similar to how we validate `x-operation-group` and other extensions.

We should create a json-schema for the namespace files and another for the category files in the `./spec` folder. Even though we currently only require our `yaml` files to be OpenAPI valid, we actually require a much stricter json-schema for these files:
-  `x-operation-group` is supported but a typo like `x-opperation-group` should not. 
-  Someone uses a new and valid OpenAPI keyword that the clients should handle accordingly but it got through the PR review without updating the Client Gen doc.
- We can enforce validation on the extensions for free through AJV.

Reply:
https://github.com/opensearch-project/opensearch-api-specification/issues/492